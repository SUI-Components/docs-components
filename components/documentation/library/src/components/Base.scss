@import './core';

#{$PREFIX} {
  color: currentColor;
  font-family: var(--ff-sui-studio-doc-sans-serif);
  font-weight: var(--fw-sui-studio-doc-regular);
  line-height: 1.4;
  @at-root {
    b#{&},
    strong#{&} {
      font-weight: var(--fw-sui-studio-doc-bold);
    }
  }
  &#{$PREFIX} {
    &-tt {
      &-uppercase {
        text-transform: var(--tt-sui-studio-doc-uppercase);
      }
      &-capitalize {
        text-transform: var(--tt-sui-studio-doc-capitalize);
      }
      &-lowercase {
        text-transform: var(--tt-sui-studio-doc-lowercase);
      }
    }
    &-fw {
      &-thin {
        font-weight: var(--fw-sui-studio-doc-thin);
      }
      &-lighter {
        font-weight: var(--fw-sui-studio-doc-lighter);
      }
      &-light {
        font-weight: var(--fw-sui-studio-doc-light);
      }
      &-regular {
        font-weight: var(--fw-sui-studio-doc-regular);
      }
      &-medium {
        font-weight: var(--fw-sui-studio-doc-medium);
      }
      &-semi-bold {
        font-weight: var(--fw-sui-studio-doc-semi-bold);
      }
      &-bold {
        font-weight: var(--fw-sui-studio-doc-bold);
      }
      &-black {
        font-weight: var(--fw-sui-studio-doc-black);
      }
    }
    &-td {
      &-underscore {
        @include td-sui-studio-doc-underscore;
      }
      &-overscore {
        @include td-sui-studio-doc-overscore;
      }
      &-underline {
        text-decoration: underline;
      }
      &-overline {
        text-decoration: overline;
      }
      &-line-through {
        text-decoration: line-through;
      }
    }
    &-deprecated {
      text-decoration: line-through;
    }
    &-full-screen {
      width: 100vw;
      min-height: 100vh;
    }
    &-full-width {
      width: 100%;
    }
    &-full-height {
      height: 100%;
    }
  }
}
