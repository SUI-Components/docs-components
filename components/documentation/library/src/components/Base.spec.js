/* global expect */
import React from 'react'
import ReactDOM from 'react-dom'

import Base, {FONT_WEIGHT, TEXT_DECORATION, TEXT_TRANSFORM} from './Base.core'

import setupEnvironment from '../helpers/setupEnvironment'

describe('components', () => {
  describe('Base', () => {
    describe('default', () => {
      const Component = Base
      const setup = setupEnvironment(Component)

      test('should render without crashing', () => {
        // Given
        const props = {}

        // When
        const component = <Component {...props} />

        // Then
        const div = document.createElement('div')
        ReactDOM.render(component, div)
        ReactDOM.unmountComponentAtNode(div)
      })

      test('should NOT render null', () => {
        // Given
        const props = {}

        // When
        const {container} = setup(props)

        // Then
        expect(container.innerHTML).toBeString()
        expect(container.innerHTML).not.toHaveLength(0)
      })

      test('should have propTypes and defaultProps ', () => {
        // Given
        const component = Component

        // When

        // Then
        expect(component.hasOwnProperty('propTypes')).toBeTruthy()
        expect(component.hasOwnProperty('defaultProps')).toBeTruthy()
      })

      test('should have proper displayName attribute', () => {
        // Given
        // const props = {}

        // When
        const component = Component

        // Then
        expect(component.displayName).toBeString()
        expect(component.displayName).not.toHaveLength(0)
        expect(component.displayName).toMatchSnapshot()
      })

      test('should display wrapping default children only', () => {
        // Given
        const props = {
          children: 'children'
        }

        // When
        const {container, getByText} = setup(props)
        const element = getByText(props.children)

        // Then
        expect(element.localName).toBe('div')
        expect(element.innerHTML).toBeString()
        expect(element.innerHTML).toBe(props.children)
        expect(container).toMatchSnapshot()
      })

      test('should render elementType if an element is given', () => {
        // Given
        const props = {
          elementType: 'element-type',
          children: 'children'
        }

        // When
        const {container, getByText} = setup(props)
        const element = getByText(props.children)

        // Then
        expect(element.localName).toBe(props.elementType)
        expect(element.innerHTML).toBeString()
        expect(element.innerHTML).toBe(props.children)
        expect(container).toMatchSnapshot()
      })

      test('should assign default element in class is given', () => {
        // Given
        const props = {
          className: 'test-classname',
          children: 'children'
        }

        // When
        const {container, getByText} = setup(props)
        const element = getByText(props.children)

        // Then
        expect(element.localName).toBe('span')
        expect(element.innerHTML).toBeString()
        expect(element.innerHTML).toBe(props.children)
        expect(container).toMatchSnapshot()
      })

      test('should assign default elementType if it is deprecated', () => {
        // Given
        const props = {
          deprecated: true,
          children: 'children'
        }

        // When
        const {container, getByText} = setup(props)
        const element = getByText(props.children)

        // Then
        expect(element.localName).toBe('span')
        expect(element.innerHTML).toBeString()
        expect(element.innerHTML).toBe(props.children)
        expect(container).toMatchSnapshot()
      })

      test('should assign default elementType if textTransform is given', () => {
        // Given
        const props = {
          textTransform: TEXT_TRANSFORM.CAPITALIZE,
          children: 'children'
        }

        // When
        const {container, getByText} = setup(props)
        const element = getByText(props.children)

        // Then
        expect(element.localName).toBe('span')
        expect(element.innerHTML).toBeString()
        expect(element.innerHTML).toBe(props.children)
        expect(container).toMatchSnapshot()
      })

      test('should assign default elementType if fontWeight is given', () => {
        // Given
        const props = {
          fontWeight: FONT_WEIGHT.BOLD,
          children: 'children'
        }

        // When
        const {container, getByText} = setup(props)
        const element = getByText(props.children)

        // Then
        expect(element.localName).toBe('span')
        expect(element.innerHTML).toBeString()
        expect(element.innerHTML).toBe(props.children)
        expect(container).toMatchSnapshot()
      })

      test('should assign default elementType if textDecoration is given', () => {
        // Given
        const props = {
          textDecoration: TEXT_DECORATION.UNDERLINE,
          children: 'children'
        }

        // When
        const {container, getByText} = setup(props)
        const element = getByText(props.children)

        // Then
        expect(element.localName).toBe('span')
        expect(element.innerHTML).toBeString()
        expect(element.innerHTML).toBe(props.children)
        expect(container).toMatchSnapshot()
      })
    })
  })
})
