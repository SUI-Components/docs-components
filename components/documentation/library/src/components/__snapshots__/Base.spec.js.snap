// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Base default should assign default element in class is given 1`] = `
<div
  id="test-container"
>
  <span
    class="sui-studio-doc sui-studio-doc-mode-light test-classname"
  >
    children
  </span>
</div>
`;

exports[`components Base default should assign default elementType if fontWeight is given 1`] = `
<div
  id="test-container"
>
  <span
    class="sui-studio-doc sui-studio-doc-fw-bold sui-studio-doc-mode-light"
  >
    children
  </span>
</div>
`;

exports[`components Base default should assign default elementType if it is deprecated 1`] = `
<div
  id="test-container"
>
  <span
    class="sui-studio-doc sui-studio-doc-deprecated sui-studio-doc-mode-light"
  >
    children
  </span>
</div>
`;

exports[`components Base default should assign default elementType if textDecoration is given 1`] = `
<div
  id="test-container"
>
  <span
    class="sui-studio-doc sui-studio-doc-td-underline sui-studio-doc-mode-light"
  >
    children
  </span>
</div>
`;

exports[`components Base default should assign default elementType if textTransform is given 1`] = `
<div
  id="test-container"
>
  <span
    class="sui-studio-doc sui-studio-doc-tt-capitalize sui-studio-doc-mode-light"
  >
    children
  </span>
</div>
`;

exports[`components Base default should display wrapping default children only 1`] = `
<div
  id="test-container"
>
  <span
    class="sui-studio-doc sui-studio-doc-mode-light"
  >
    children
  </span>
</div>
`;

exports[`components Base default should have proper displayName attribute 1`] = `"Base"`;

exports[`components Base default should render elementType if an element is given 1`] = `
<div
  id="test-container"
>
  <element-type
    classname="sui-studio-doc sui-studio-doc-mode-light"
  >
    children
  </element-type>
</div>
`;
